src/Messages/OWSMessageSender.m

```diff
@@ -697,26 +677,12 @@ - (void)groupSend:(NSArray<SignalRecipient *> *)recipients
     [completionFuture catchDo:^(id failure) {
         // failure from toc_thenAll yeilds an array of failed Futures, rather than the future's failure.
         if ([failure isKindOfClass:[NSArray class]]) {
-            NSArray *errors = (NSArray *)failure;
-            for (TOCFuture *failedFuture in errors) {
-                if (!failedFuture.hasFailed) {
-                    // If at least one send succeeded, don't show message as failed.
-                    // Else user will tap-to-resend to all recipients, including those that already received the
-                    // message.
-                    return successHandler();
-                }
-            }
-
-            // At this point, all recipients must have failed.
-            // But we have all this verbose type checking because TOCFuture doesn't expose type information.
-            id lastError = errors.lastObject;
-            if ([lastError isKindOfClass:[TOCFuture class]]) {
-                TOCFuture *failedFuture = (TOCFuture *)lastError;
-                if (failedFuture.hasFailed) {
-                    id failureResult = failedFuture.forceGetFailure;
+            NSArray *groupSendFutures = (NSArray *)failure;
+            for (TOCFuture *groupSendFuture in groupSendFutures) {
```

Comment:
@michaelkirk Hmm, looking over this last commit, I'm having second thoughts about whether moving the `isRetryable` flag from being a block parameter to a NSError property was a good idea.  Originally I thought it'd be a good idea for two reasons:

* I thought I might split it out into separate `isRetryableForContacts` and `isRetryableForGroups` flags, but my intuition was wrong - I don't think there aren't any scenarios where we want to retry for contacts but not for groups or vice versa.  
* It'd be consistent with the `shouldBeIgnoredForGroups` property.  But since we're using that in so few place,s who cares?

Anyway, I might revert most of this commit.